source/features/faster-pr-diff-options.tsx

```diff
@@ -0,0 +1,94 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import * as icons from '../libs/icons';
+import features from '../libs/features';
+
+function createDiffStyleToggle() {
+	const params = new URLSearchParams(location.search);
+	const isUnified = select.exists([
+		'[value="unified"][checked]', // Form in PR
+		'.table-of-contents .selected[href$=unified]' // Link in single commit
+	].join());
+
+	const makeLink = (type, icon, selected) => {
+		params.set('diff', type);
+		return <a
+			className={`btn btn-sm BtnGroup-item tooltipped tooltipped-s ${selected ? 'selected' : ''}`}
+			aria-label={`Show ${type} diffs`}
+			href={`?${params}`}>
+			{icon}
+		</a>;
+	};
+
+	return <>
+		{makeLink('unified', icons.diff(), isUnified)}
+		{makeLink('split', icons.book(), !isUnified)}
+	</>;
+}
+
+function createWhitespaceButton() {
+	const searchParams = new URLSearchParams(location.search);
+	const isHidingWhitespace = searchParams.get('w') === '1';
+
+	if (isHidingWhitespace) {
+		searchParams.delete('w');
+	} else {
+		searchParams.set('w', '1');
+	}
+
+	return (
+		<a href={`?${searchParams}`}
+			data-hotkey="d w"
+			className={`btn btn-sm btn-outline tooltipped tooltipped-s ${isHidingWhitespace ? 'bg-gray-light text-gray-light' : ''}`}
+			aria-label={`${isHidingWhitespace ? 'Show' : 'Hide'} whitespace in diffs`}>
+			{isHidingWhitespace ? icons.check() : false} No Whitespace
+		</a>
+	);
+}
+
+function wrap(...elements) {
+	if (features.isSingleCommit()) {
+		return <div className="float-right">
+			{...elements.map(element => <div className="ml-3 BtnGroup">{element}</div>)}
+		</div>;
+	}
+
+	return <>{elements.map(element => <div className="diffbar-item">{element}</div>)}</>;
```

Comment:
This function is ugly, but I already rewrote it a few times. If you think of another way to write this, please write it.

Reply:
Considering the alternatives I can think of, this doesn't look so bad.